lib/assert.js

```diff
@@ -750,4 +756,14 @@ assert.strict = Object.assign(strict, assert, {
   notEqual: assert.notStrictEqual,
   notDeepEqual: assert.notDeepStrictEqual
 });
+
+Object.defineProperty(assert, 'undefinedProperty', {
+  value: Symbol('node.assert.undefined-property'),
```

Comment:
Since it’s not a global Symbol, I’d strongly prefer naming it `assert.undefinedProperty`.